src/AutoRest.CSharp/Common/Generation/Writers/RequestWriterHelpers.cs

```diff
@@ -208,6 +208,7 @@ public static void WriteRequestCreation(CodeWriter writer, RestClientMethod clie
 
                 if (writeUserAgentOverride)
                 {
+                    writer.Line($"{request}.Headers.SetValue(HttpHeader.Names.UserAgent, _userAgent);");
```

Comment:
We shouldn't need this here this is what the MgmtTelemitryPolicy does for us

Reply:
Yeah, I understand that, but the reason that we need this line is `MgmtTelemetryPolicy` is going to extract the `User-Agent` information from [message's property](https://github.com/Azure/azure-sdk-for-net/blob/e09905cbba81cd31a4add5bb4425127b9dc1589f/sdk/resourcemanager/Azure.ResourceManager/src/Internal/MgmtTelemetryPolicy.cs#L21), however we just pass the request to the OperationInternals, which doesn't contains this information. So this line is just used to store the 'User-Agent' so that we can pull out it in OperationInternals.